src/test/java/com/yahoo/bullet/kafka/KafkaSubscriberTest.java

```diff
@@ -156,15 +142,15 @@ public void testRateLimit() throws PubSubException, InterruptedException {
         KafkaConsumer<String, byte[]> consumer = (KafkaConsumer<String, byte[]>) mock(KafkaConsumer.class);
         ConsumerRecords<String, byte[]> records = makeConsumerRecords(randomID, new PubSubMessage(randomID, randomMessage, null));
         when(consumer.poll(any())).thenReturn(records);
-        KafkaSubscriber subscriber = new KafkaSubscriber(consumer, 20, 5, 10L);
+        KafkaSubscriber subscriber = new KafkaSubscriber(consumer, 20, 5, 50L);
 
         for (int i = 0; i < 5; i++) {
             Assert.assertNotNull(subscriber.receive());
         }
         Assert.assertNull(subscriber.receive());
 
         // Sleep to reset interval
-        Thread.sleep(15);
+        Thread.sleep(55);
```

Comment:
I don't think we should do 5 messages and low timeouts like these. These are gonna be flaky on whatever slow machine we run. We should probably do a binary thing or at least check if messages emitted are <= rate limit. It would also be nice if we slept as little as possible - 5-10ms at most

Reply:
Oh.. that sounds a lot smarter. Is it ok if I let it spin for 5 seconds then?